lxd/instance/drivers/driver_qemu.go

```diff
@@ -1363,6 +1369,9 @@ func (d *qemu) start(stateful bool, op *operationlock.InstanceOperation) error {
 		}
 	}()
 
+	// Add allocated qemu vhost file descriptor
```

Comment:
I think we need to move this line and the fdFiles definition and revert bits above to where this line is:

```
vsockID, vsockF, err := d.nextVsockID()
```

Otherwise there is plenty of room for the vsockF to be returned and then this function return with an error before the defer that closes the file handle is setup. This would cause a file descriptor leak.

Reply:
Moved to the right place.